src/browser/screen.js

```diff
@@ -204,16 +411,45 @@ function ScreenAdapter(options, screen_fill_buffer)
         return image;
     };
 
-    this.put_char = function(row, col, chr, blinking, bg_color, fg_color)
+    /**
+     * Return content of the complete text screen as an array of Unicode strings.
+     * Trim each line's trailing whitespace unless keep_whitespace is true.
+     * If defined, codepage must be an array of 256 single-character strings.
+     * If codepage is undefined or null an internal US-codepage (CP437) is used.
+     *
+     * @param {boolean|undefined} keep_whitespace
+     * @param {Array<string>|null|undefined} codepage
+     * @return {Array<string>} The screen's current text rows.
+     */
+    this.grab_text_content = function(keep_whitespace, codepage)
+    {
+        const text_rows = [];
+        if(mode === MODE_TEXT || mode === MODE_GRAPHICAL_TEXT)
+        {
+            codepage = codepage || charmap;
+            for(var row = 0, i_chr = 0; row < text_mode_height; row++)
+            {
+                let line = "";
+                for(var col = 0; col < text_mode_width; col++, i_chr += TEXT_BUF_COMPONENT_SIZE)
+                {
+                    line += codepage[text_mode_data[i_chr]];
+                }
+                text_rows.push(keep_whitespace ? line : line.trimEnd());
+            }
+        }
+        return text_rows;
+    };
```

Comment:
Could you merge this function into get_text_row/get_text_screen. I'd suggest removing the `keep_whitespace` parameter, the caller could do that if they want.

Reply:
Done, resolved in 9f2d029 and e58e02b.